scico/linop/xray/astra.py

```diff
@@ -156,17 +116,10 @@ def __init__(
         else:
             raise ValueError(f"Invalid device specified; got {device}.")
 
-        if self.num_dims == 3 and self.device == "cpu":
-            raise ValueError("No CPU algorithm for 3D projection.")
-
-        if self.num_dims == 3:
-            # not needed for astra's 3D algorithm
-            self.proj_id = None
-        elif self.num_dims == 2:
-            if self.device == "cpu":
-                self.proj_id = astra.create_projector("line", self.proj_geom, self.vol_geom)
-            elif self.device == "gpu":
-                self.proj_id = astra.create_projector("cuda", self.proj_geom, self.vol_geom)
+        if self.device == "cpu":
+            self.proj_id = astra.create_projector("line", self.proj_geom, self.vol_geom)
+        elif self.device == "gpu":
+            self.proj_id = astra.create_projector("cuda", self.proj_geom, self.vol_geom)
```

Comment:
Does 2D projection on GPU work? Does it work with astra's FBP? The astra docs say "NB: This functionality has not yet been implemented everywhere. This projector is currently only usable in astra_create_sino and astra_create_backprojection." https://astra-toolbox.com/docs/proj2d.html#miscellaneous.

Reply:
No idea. This PR doesn't make any changes to the 2D projection - it just pulled out the 3D projection as a separate class.